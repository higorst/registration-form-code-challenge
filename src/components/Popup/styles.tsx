import styled from 'styled-components/native'
import { Dimensions, ViewProps } from 'react-native'
import { Colors } from '../../styles/Colors'

export const Message = styled.Text`
    font-size: 20px;
    text-align: center;

    margin-left: 10px;
    margin-right: 10px;
    margin-top: 40px;
    margin-bottom: 20px;
    max-width: ${`${Dimensions.get('window').width * 0.6}px`};

    font-weight: bold;
    color: ${Colors.backgroundApp};
`