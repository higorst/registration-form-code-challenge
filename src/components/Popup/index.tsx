import React from 'react'
import { Colors } from '../../styles/Colors';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import RegexControllers from '../../controllers/Regexcontrollers';

import Dialog, { DialogFooter, SlideAnimation, DialogButton, DialogContent } from 'react-native-popup-dialog'
import { Message } from './styles'

// Redux
import { connect } from "react-redux";
import { SAGA_DELETE } from '../../redux/sagas/types';

import { Constants } from '../../constants';

interface Popuptype {
    visible: boolean
    user_name: string

    onCancel: any
    onOk: any
}
const Popup = (props: Popuptype) => {

    return (
        <Dialog
            visible={props.visible}
            onTouchOutside={props.onCancel}
            dialogAnimation={new SlideAnimation({
                slideFrom: 'bottom',
            })}
            footer={
                <DialogFooter>
                    <DialogButton
                        text="CANCEL"
                        onPress={props.onCancel}
                    />
                    <DialogButton
                        text="OK"
                        onPress={props.onOk}
                    />
                </DialogFooter>
            }
        >
            <DialogContent>
                <Message>Deseja realmente excluir o usu√°rio {props.user_name}?</Message>
            </DialogContent>
        </Dialog>
    )
}

export default Popup