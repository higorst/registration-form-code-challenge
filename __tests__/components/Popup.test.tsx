import React from 'react';
import renderer from 'react-test-renderer';
import Popup from '../../src/components/Popup/index'

jest.useFakeTimers()
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper')

describe('<Popup />', () => {
    it('renders Popup Component correctly', () => {
        const tree = renderer.create(
            <Popup
                visible={true}
                onCancel={() => {}}
                user_name='Snapshot'
                onOk={() => {}}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});