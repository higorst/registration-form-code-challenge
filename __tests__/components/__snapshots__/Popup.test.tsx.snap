// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> renders Popup Component correctly 1`] = `null`;
